nifi-commons/nifi-utils/src/main/java/org/apache/nifi/util/FormatUtils.java

```diff
@@ -425,4 +434,32 @@ public static String formatNanos(final long nanos, final boolean includeTotalNan
 
         return sb.toString();
     }
+
+    public static DateTimeFormatter prepareLenientCaseInsensitiveDateTimeFormatter(String pattern) {
+        return new DateTimeFormatterBuilder()
+                .parseLenient()
+                .parseCaseInsensitive()
+                .appendPattern(pattern)
+                .toFormatter(Locale.US);
```

Comment:
In general it make sense but for some reasons someone hardcoded US locales in all places in nifi expression language. Take a look at `DateCastEvaluator`, `FormatEvaluator`, `StringToDateEvaluator` before my change. Everywhere was used `new SimpleDateFormat(formatValue, Locale.US)`. And it is quite good for most cases because quite often names of months, day of weeks and so on used in data are in English even if locales are different.

Reply:
Thanks for pointing out the existing implementation references to `Locale.US`, that is worth noting.  In that case, this seems fine for now, perhaps worth revisiting in the future.